lib/gollum/app.rb

```diff
@@ -230,14 +230,15 @@ class App < Sinatra::Base
         halt 500 unless tempfile.is_a? Tempfile
 
         if wiki.per_page_uploads
-          # remove base_url and gollum/* subpath if necessary
-          dir = request.referer.
-                  sub(request.base_url, '').
-                  sub(/.*gollum\/[-\w]+\//, '')
+          dir = request.referer.sub(request.base_url, '')
           # remove base path if it is set
-          dir = dir.sub(wiki.base_path, '') if wiki.base_path
+          dir.sub!(/^#{wiki.base_path}/, '') if wiki.base_path
```

Comment:
Why the added `'/'`? Does this work when `base_path` already ends with a slash?

Reply:
`/this/` is not a string but a regex. With the leading `^`, I just wanted to ensure that only `base_path` at the beginning is replaced.

Reply:
Aah sorry for misreading!